framework/framework.go

```diff
@@ -97,13 +98,43 @@ func (f *Framework) ProcessCreate(obj interface{}, resources ...Resource) error
 	return nil
 }
 
+// ProcessCreateWithBackoff is the same as ProcessCreate but takes an additional
+// backoff factory returning new backoff implementations to retry the creation
+// on resource errors. Retries are implemented using a retry resource that wraps
+// each given resource. Errors are logged eventually in case an appropriate
+// logger is configured in the operator framework.
+func (f *Framework) ProcessCreateWithBackoff(obj interface{}, resources []Resource, backoffFactory func() backoff.BackOff) error {
+	var retryResources []Resource
+
+	for _, r := range resources {
+		resourceConfig := DefaultRetryResourceConfig()
+		resourceConfig.BackOff = backoffFactory()
+		resourceConfig.Logger = f.logger
+		resourceConfig.Resource = r
+
+		retryResource, err := NewRetryResource(resourceConfig)
```

Comment:
> Why the backoff can't be applied in the Process* methods around resource methods invocations instead of creating a new resource type?

In the Process* methods 4 different methods are called. In case the last one throws an error all former method calls are repeated if we wrap it into one backoff operation. That felt very wrong to me because an operation should only be retried in case it fails. Other operations should be untouched. We could get rid of the retry resource if we wrap each and every method into its own backoff operation. I thought it is better to implement the interface we have to have the whole design more composable. 

> I don't like current approach, because to use retry you have to rewrite all resources.

I cannot follow. You do not need to rewrite anything. The retry resource is just used under the hood of the Process*WithBackoff methods. You only have to rename these functions and add the backoff factory, which you would need to do anyway with every other solution.